rabbitmq-sever build error caused by rabbitmqctl does not support Elixir 1.13.1

### Describe the bug
rabbitmq-sever build error caused by rabbitmqctl does not support Elixir 1.13.1

### Steps To Reproduce
1. nix-shell -p rabbitmq-server

```console 
[user@system:~]$ nix-shell -p rabbitmq-server
...
Generated parallel_stream app
==> json
Compiling 13 files (.ex)
Generated json app
==> recon
Compiling 6 files (.erl)
Generated recon app
==> csv
Compiling 10 files (.ex)
Generated csv app
==> observer_cli
Compiling 13 files (.erl)
Generated observer_cli app
==> rabbitmqctl
** (Mix) You're trying to run :rabbitmqctl on Elixir v1.13.1 but it has declared in its mix.exs file it supports only Elixir >= 1.10.4 and < 1.13.0
make[2]: *** [Makefile:106: escript/rabbitmqctl] Error 1
make[2]: Leaving directory '/private/tmp/nix-build-rabbitmq-server-3.9.8.drv-1/rabbitmq-server-3.9.8/deps/rabbitmq_cli'
make[1]: *** [../../erlang.mk:4512: deps] Error 2
make[1]: Leaving directory '/private/tmp/nix-build-rabbitmq-server-3.9.8.drv-1/rabbitmq-server-3.9.8/deps/rabbit'
make: *** [erlang.mk:4512: deps] Error 2
error: builder for '/nix/store/9b3njwkp6vq2s6a6xj3981bm9bbrn20y-rabbitmq-server-3.9.8.drv' failed with exit code 2
```

### Expected behavior
rabbitmq-sever built and installed on the nix shell.

### Additional context
`elixir 1.13.1` was released in [December 14th](https://github.com/elixir-lang/elixir/releases/tag/v1.13.1) and `elixir nixpkg` was updated in [December 17th](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/interpreters/elixir/1.13.nix) but [RabbitMQ's rabbitmq-sever does not yet support Elixir 1.13.1](https://github.com/rabbitmq/rabbitmq-server/blob/4993fefa08fa0785385d23465d1f6de20bd4649c/deps/rabbitmq_cli/mix.exs#L14).
I am very new to NixOS project, so I am not sure how these conflicts are commonly resolved. I tried to build `rabbitmq-sever` with `elixir_1_12` and it built it successfully, just replacing `elixir` with `elixir_1_12` in rabbitmq_server'd [default.nix](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/servers/amqp/rabbitmq-server/default.nix).

### Notify maintainers
@turion 

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.1.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4pre20210601_5985b8b`
 - channels(user): `"nixpkgs-22.05pre340506.5c37ad87222"`
 - nixpkgs: `/Users/user/.nix-defexpr/channels/nixpkgs`
```

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -21675,6 +21675,7 @@ with pkgs;
 
   rabbitmq-server = callPackage ../servers/amqp/rabbitmq-server {
     inherit (darwin.apple_sdk.frameworks) AppKit Carbon Cocoa;
+    elixir = elixir_1_12;
   };
 
   radicale1 = callPackage ../servers/radicale/1.x.nix { };

